nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageServiceImpl.java

```diff
@@ -374,149 +409,166 @@ public DocumentModel restoreContentFromColdStorage(CoreSession session, Document
-            BlobStatus blobStatus = getBlobStatus(doc);
-            if (blobStatus == null) {
-                continue;
-            }
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String username = SecurityConstants.SYSTEM_USERNAME;
+        String commandId = bulkService.submit(new BulkCommand.Builder(
+                PropagateRestoreFromColdStorageContentAction.ACTION_NAME, query, username).build());
 
-            if (blobStatus.isDownloadable()) {
-                available++;
-                beingRetrieved--;
-
-                // Check if the Document should be restored definitively
-                Serializable undoMove = doc.getPropertyValue(COLD_STORAGE_TO_BE_RESTORED_PROPERTY);
-                if (Boolean.TRUE.equals(undoMove)) {
-                    restoreMainContent(doc);
-                    doc.setPropertyValue(COLD_STORAGE_TO_BE_RESTORED_PROPERTY, false);
-                    session.saveDocument(doc);
-                } else {
-                    doc.setPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY, false);
-                    session.saveDocument(doc);
-
-                    DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
-                    Instant downloadableUntil = blobStatus.getDownloadableUntil();
-                    if (downloadableUntil != null) {
-                        ctx.getProperties()
-                           .put(COLD_STORAGE_CONTENT_AVAILABLE_UNTIL_MAIL_TEMPLATE_KEY, downloadableUntil.toString());
-                        doc.setPropertyValue(COLD_STORAGE_CONTENT_DOWNLOADABLE_UNTIL, Date.from(downloadableUntil));
-                        session.saveDocument(doc);
-                    }
-                    String fileName = ((Blob) doc.getProperty(COLD_STORAGE_CONTENT_PROPERTY).getValue()).getFilename();
-                    String serverUrl = NotificationServiceHelper.getNotificationService().getServerUrlPrefix();
-                    String downloadUrl = serverUrl
-                            + downloadService.getDownloadUrl(doc, COLD_STORAGE_CONTENT_PROPERTY, fileName);
-                    ctx.getProperties().put(COLD_STORAGE_CONTENT_ARCHIVE_LOCATION_MAIL_TEMPLATE_KEY, downloadUrl);
-                    eventService.fireEvent(ctx.newEvent(COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME));
-                }
-            } else if(!blobStatus.isOngoingRestore()) {
-                // the blob was probably retrieved and it already went back to cold storage
-                // Let's flag it as not being retrieved
-                log.debug("Document {} is flagged as being retrieved but not its blob", doc::getPath);
-                beingRetrieved--;
-                doc.setPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY, false);
-                session.saveDocument(doc);
-            }
+        BulkStatus status = bulkService.getStatus(commandId);
+        if (status == null) {
```

Comment:
I think this check is always false, `getStatus` never returns null.

Reply:
I removed this part of the code